cloudwatch-kinesis/firehose-delivery-stream.tf

```diff
@@ -0,0 +1,153 @@
+data "aws_lb" "main" {
+  name = local.ecs_cluster_name
+}
+
+# https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SubscriptionFilters.html#FirehoseExample
+resource "aws_iam_role" "kinesis" {
+  name = "kinesis-${local.name}"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "firehose.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "kinesis" {
+  name = "kinesis-policy-${local.name}"
+
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Action" : [
+          "s3:AbortMultipartUpload",
+          "s3:GetBucketLocation",
+          "s3:GetObject",
+          "s3:ListBucket",
+          "s3:ListBucketMultipartUploads",
+          "s3:PutObject"
+        ],
+        "Effect" : "Allow",
+        "Resource" : [
+          var.log_bucket_arn,
+          "${var.log_bucket_arn}/*"
+        ]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "logs:DescribeLogGroups",
+          "logs:DescribeLogStreams",
+          "logs:GetLogEvents",
+          "logs:PutLogEvents"
+        ],
+        "Resource" : "*"
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "kms:Encrypt",
+          "kms:Decrypt",
+          "kms:GenerateDataKey"
+        ],
+        # TODO: Please stricten the rules
+        "Resource" : "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "kinesis" {
+  role       = aws_iam_role.kinesis.name
+  policy_arn = aws_iam_policy.kinesis.arn
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "main" {
```

Comment:
Should we add a 2nd (optional) stream to also send all logs to a S3 bucket for long term storage? @samkahchiin (using the `extended_s3` endpoint)

Reply:
ahhh. This comment is hidden in the front page.
I am thinking about this as well. 
It can be achieved by setting the mode to AllData, so we don't need the extended_s3 endpoint

Reply:
That sounds good to me